### PR TITLE
Add version 2018-06-01 with pricing configuration (second time)

### PR DESCRIPTION
**Pull request by amireluk**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by amireluk** [comment]
2nd iteration of this PR which was closed - https://github.com/Azure/azure-rest-api-specs/pull/4072
With slight change to the PUT operation to reflect the API.

**Comment by orparnes** [comment]
@chlahav is added to the review. #Closed

**Comment by amireluk** [comment]
@kpajdzik  Can you please review the PR ?

**Comment by sergey-shandar** [comment]
@praries880 see also this PR https://github.com/Azure/azure-rest-api-specs/pull/4072

**Comment by sergey-shandar** [comment]
@amireluk have you addressed ARM issues (see https://github.com/Azure/azure-rest-api-specs/pull/4072)? 

@simonjaeger let me know if you are ok with the changes.

**Comment by amireluk** [comment]
> @amireluk have you addressed ARM issues (see #4072)?
> 
> @simongdavies let me know if you are ok with the changes.

ARM issues in #4072 was about long running operation header. In this iteration the operation is not a long running operation. It returns 200 OK.

**Review by sergey-shandar — APPROVED** [review]

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: do you support pagination?

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: Isnt this similar notion as a SKU? If yes, it should be modeled using the sku property which is a top level property. More details can be found in ARM Resource provider contract.

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: Did you had a meeting earlier with Simon about this new resource type? I wanted to get some more context on the usecase for the pricing resource type. what scenario is it enabling.

**Review by ravbhatnagar — COMMENTED** [review]
@amireluk - some comments

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json
Comment (amireluk) [reply specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: no, the number of returned items is limited.

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json
Comment (amireluk) [reply specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: Please note it's an existing type, I'm adding a new API version. The previous version is already documented.

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json
Comment (amireluk) [reply specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: It's a new API version to an existing type, customers are using it and used to use it. I don't think we want to deprecate it and use the SKU propery instead as it will be very confusing to existing customers.
I'm also not sure we can use the SKU property even if we wanted to since this pricing type implies the tier of the subscritpion in Microsoft.Secruity, and not the sku of a specific resource.

**Comment by ravbhatnagar** [comment]
@amireluk  - Got it, it was hard to review using the diff tool and know what changed. For next time, please remember that when adding new api-version, the first commit should be the copy of the previous api-version. and then changes to the new api-version should come in subsequent commits. it will make it easier to review.
It looks like you are just moving /pricings resourceType to a non-preview api-version. And there are no changes to this existing resource type from preview. Is that correct. 
Also, this means the customer will not be able to use the other resource types like securityContacts etc in this new api-version. Is that the intention?

**Comment by amireluk** [comment]
> Also, this means the customer will not be able to use the other resource types like securityContacts etc in this new api-version. Is that the intention?

The non-preview pricing API is slightly different than the preview one, that's the reason for the version change.
I understand that this API version will only include this resource. The other resources have not changed.

**Comment by KrisBash** [comment]
> I understand that this API version will only include this resource. The other resources have not changed.

@amireluk Can you please bring forward the unchanged APIs into this version? It creates a degraded user experience if different resources in the same namespace require different API versions.

**Comment by sergey-shandar** [comment]
@KrisBash @amireluk any progress on this PR?

**Review by chlahav — APPROVED** [review]

**Comment by chlahav** [comment]
@KrisBash we want to split our swagger files to types (similar to Compute) to mitigate the degraded user experience.
we will do that right after this merge will happen to avoid merge conflicts.
is there anything additional that we need to do to get the PR approval?

**Pull request merged**
